sweepbatcher/sweep_batch.go

```diff
@@ -0,0 +1,1390 @@
+		)
+		if err != nil {
+			return err
+		}
+
+		// Sanity check that we have all the nonces.
+		if !haveAllNonces {
+			return fmt.Errorf("invalid MuSig2 session: " +
+				"nonces missing")
+		}
+
+		var digest [32]byte
+		copy(digest[:], sigHash)
+
+		// Since our MuSig2 session has all nonces, we can now create
+		// the local partial signature by signing the sig hash.
+		_, err = b.signerClient.MuSig2Sign(
+			ctx, musig2SessionInfo.SessionID, digest, false,
+		)
+		if err != nil {
+			return err
+		}
+
+		// Now combine the partial signatures to use the final combined
+		// signature in the sweep transaction's witness.
+		haveAllSigs, finalSig, err := b.signerClient.MuSig2CombineSig(
+			ctx, musig2SessionInfo.SessionID, [][]byte{serverSig},
+		)
+		if err != nil {
+			return err
+		}
+
+		if !haveAllSigs {
+			return fmt.Errorf("failed to combine signatures")
+		}
+
+		// To be sure that we're good, parse and validate that the
+		// combined signature is indeed valid for the sig hash and the
+		// internal pubkey.
+		err = b.verifySchnorrSig(
+			htlcScript.TaprootKey, sigHash, finalSig,
+		)
+		if err != nil {
+			return err
+		}
+
+		packet.UnsignedTx.TxIn[i].Witness = wire.TxWitness{
+			finalSig,
+		}
+	}
+
+	return nil
+}
+
+// updateRbfRate updates the fee rate we should use for the new batch
+// transaction. This fee rate does not guarantee RBF success, but the continuous
+// increase leads to an eventual successful RBF replacement.
+func (b *batch) updateRbfRate(ctx context.Context) error {
+	// If the feeRate is unset then we never published before, so we
+	// retrieve the fee estimate from our wallet.
+	if b.rbfCache.FeeRate == 0 {
+		b.log.Infof("initializing rbf fee rate for conf target=%v",
+			b.Cfg.BatchConfTarget)
+		rate, err := b.wallet.EstimateFeeRate(
+			ctx, b.Cfg.BatchConfTarget,
+		)
+		if err != nil {
+			return err
+		}
+
+		// Set the initial value for our fee rate.
+		b.rbfCache.FeeRate = rate
+	} else {
+		// Bump the fee rate by the configured step.
+		b.rbfCache.FeeRate += defaultFeeRateStep
+	}
+
+	b.rbfCache.LastHeight = b.currentHeight
+
+	return b.persist(ctx)
+}
+
+// monitorSpend monitors the primary sweep's outpoint for spends. The reason we
+// monitor the primary sweep's outpoint is because the primary sweep was the
+// first sweep that entered this batch, therefore it is present in all the
+// versions of the batch transaction. This means that even if an older version
+// of the batch transaction gets confirmed, due to the uncertainty of RBF
+// replacements and network propagation, we can always detect the transaction.
+func (b *batch) monitorSpend(ctx context.Context, primarySweep sweep) {
+	spendCtx, cancel := context.WithCancel(ctx)
+
+	spendChan, spendErr, err := b.chainNotifier.RegisterSpendNtfn(
+		spendCtx, &primarySweep.outpoint, primarySweep.htlc.PkScript,
+		primarySweep.initiationHeight,
+	)
+	if err != nil {
+		b.writeToErrChan(ctx, err)
```

Comment:
> There's no need to call cancel here
Linter complains about `cancel` not being used on all paths. I know that the context will effectively exit because of the parent context exiting, but can't tell the linter that

Reply:
Ah right right, linter being too keen to lint.